### PR TITLE
Docs/optimize route group semantics

### PR DESCRIPTION
**Pull request by WinmezzZ**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 97ed460</samp>

Updated the documentation of the `routes` feature in `modern.js` for both English and Chinese languages. Explained the new concepts and API of "Route Grouping" and "Route Flating" and their benefits for developers.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 97ed460</samp>

*  Rename and rewrite section titles and descriptions to clarify the features of using `__` and `$` in route files ([link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L250-R250), [link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L256-R256), [link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134L269-R271), [link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2L275-R277))
*  Add sentences to highlight the benefit of using `$` to create routes without layout components or deep directory levels ([link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-d241f975611226b5855634d8b23b1eea196b7f7657e4ae1ca82ff7ee8ce4a134R293-R294), [link](https://github.com/web-infra-dev/modern.js/pull/4957/files?diff=unified&w=0#diff-00505071a8d949d0c52469c79e789619ec5e076f90f34b350af62959fe5aaea2R299-R300))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by KyrieLii** [comment]
@WinmezzZ 感谢贡献，不过请先签署下 CLA 

Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/web-infra-dev/modern.js?pullRequest=4957) before we can accept your contribution.

**Comment by WinmezzZ** [comment]
done

File: packages/document/main-doc/docs/zh/guides/basic-features/routes.mdx [hunk 1]
```diff
@@ -272,9 +272,9 @@ params['*']; // => 'aaa/bbb'
 
 Modern.js 会生成 `/login` 和 `/sign` 两条路由，`__auth/layout.tsx` 组件会作为 `login/page.tsx` 和 `signup/page.tsx` 的布局组件，但`__auth` 不会作为路由路径片段。
 
-当需要为某些类型的路由，做独立的布局，或是想要将路由做归类时，这一功能非常有用。
+当需要为某些类型的路由，做独立的布局，或是想要将路由做归类和分组时，这一功能非常有用。
 
-### 无布局路径
+### 路由扁平化
```
Comment (zllkjc) [inline packages/document/main-doc/docs/zh/guides/basic-features/routes.mdx:277]: 感觉上面部分「路由分组」比较确切，扁平化看起来有些奇怪。虽然之前的也不好理解。还有别的想法吗，有点类似于文件目录压缩的感觉

File: packages/document/main-doc/docs/en/guides/basic-features/routes.mdx [hunk 2]
```diff
@@ -266,9 +266,9 @@ When the directory name starts with `__`, the directory name will not be convert
 
 Modern.js will generate two routes, `/login` and `/signup`. The `__auth/layout.tsx` component will serve as the layout component for `login/page.tsx` and `signup/page.tsx`, but `__auth` will not be a route path segment.
 
-This feature is very useful when you need to create independent layouts for certain types of routes or want to classify routes.
+This feature is very useful when you need to create independent layouts or group routes for certain types of routes or want to classify routes.
 
-### No Layout
+###  Route Flating
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/basic-features/routes.mdx:271]: One more space

File: packages/document/main-doc/docs/zh/guides/basic-features/routes.mdx (hunk 1 above)
Comment (WinmezzZ) [reply packages/document/main-doc/docs/zh/guides/basic-features/routes.mdx:277]: 可能中文的意思表达不太合适？但是这个概念就是称为 flat 

https://github.com/kiliman/remix-flat-routes

**Pull request closed without merging**